src/http-api.js

```diff
@@ -679,6 +686,11 @@ module.exports.MatrixHttpApi.prototype = {
                     handlerFn(err, response, body);
                 },
             );
+            req.onprogress = (e) => {
```

Comment:
I think you should comment that this will only work in the browser (where `opts.request` is `browser-request` instead of `request`).

Indeed, I'm not sure about the morality of setting `onprogress` attributes on the `request` object. Perhaps this should be protected by `if ('onprogress' in req)` or something?